code/_onclick/ai.dm

```diff
@@ -71,8 +75,14 @@
 	*/
 	if(INVOKE_EVENT(src, /event/uattack, "atom" = A)) //This returns 1 when doing an action intercept
 		return
-	A.add_hiddenprint(src)
-	A.attack_ai(src)
+	
+	if(istype(current, /obj/machinery/turret))
+		var/obj/machinery/turret/T = current
+		if(T.enabled && T.raised)
+			T.shootAt(A)
```

Comment:
where's the cooldown eggman? because it's not in turret/shootAt

you going to let the AI shoot lasers and electrodes as fast as it can click?

Reply:
The cooldown is in the shootAt proc